pybop/plot/standard_plots.py

```diff
@@ -90,68 +91,16 @@ def __init__(
         if trace_options:
             self.trace_options.update(trace_options)
 
-        # Check trace_names and set attribute
-        if isinstance(trace_names, str):
-            self.trace_names = [trace_names]
-        else:
-            self.trace_names = trace_names
-
-        # Check type and dimensions of data
-        # What we want is a list of 'things plotly can take', e.g. numpy arrays or lists of numbers
-        if isinstance(self.x, list):
-            # If it's a list of numpy arrays, it's fine
-            # If it's a list of lists, it's fine
-            # If it's neither, it's a list of numbers that we need to wrap
-            if not isinstance(self.x[0], np.ndarray) and not isinstance(
-                self.x[0], list
-            ):
-                self.x = [self.x]
-        elif isinstance(self.x, np.ndarray):
-            self.x = np.squeeze(self.x)
-            if self.x.ndim == 1:
-                self.x = [self.x]
-            else:
-                self.x = self.x.tolist()
-        if isinstance(self.y, list):
-            if not isinstance(self.y[0], np.ndarray) and not isinstance(
-                self.y[0], list
-            ):
-                self.y = [self.y]
-        if isinstance(self.y, np.ndarray):
-            self.y = np.squeeze(self.y)
-            if self.y.ndim == 1:
-                self.y = [self.y]
-            else:
-                self.y = self.y.tolist()
-        if len(self.x) > 1 and len(self.x) != len(self.y):
-            raise ValueError(
-                "Input x should have either one data series or the same number as y."
-            )
-
         # Attempt to import plotly when an instance is created
         self.go = PlotlyManager().go
 
         # Create layout
         if self.layout is None:
             self.layout = self.go.Layout(**self.layout_options)
 
-        # Wrap trace names
-        if self.trace_names is not None:
-            for i, name in enumerate(self.trace_names):
-                self.trace_names[i] = self.wrap_text(name, width=self.trace_name_width)
-
-        # Create a trace for each trajectory
-        self.traces = []
-        x = self.x[0]
-        for i in range(0, len(self.y)):
-            if len(self.x) > 1:
-                x = self.x[i]
-            if self.trace_names is not None:
-                self.trace_options["name"] = self.trace_names[i]
-            else:
-                self.trace_options["showlegend"] = False
-            trace = self.create_trace(x, self.y[i], **self.trace_options)
-            self.traces.append(trace)
+        # Add traces
+        if self.x is not None:
```

Comment:
Probably a good idea to make sure both objects are provided:

```suggestion
        if x and y is not None:
```